### PR TITLE
Source mapping url digest

### PR DESCRIPTION
**Pull request by Szetty**
#1936 solved, also I think we will need to digest the contents from *.map files too, I can do that too, but I need a feedback if it's okay

### CODE REVIEW

File: lib/phoenix/digester.ex [hunk 1]
```diff
@@ -185,35 +185,43 @@ defmodule Phoenix.Digester do
   end
 
   defp digested_contents(file, manifest) do
-    if Path.extname(file.filename) == ".css" do
-      digest_asset_references(file, manifest)
-    else
-      file.content
+    case Path.extname(file.filename) do
```
Comment (josevalim) [inline lib/phoenix/digester.ex]: I think we have two options. Or we pattern match here and then call `css_asset_references` and `js_asset_references` or we call `digest_asset_references(Path.extname(file.filename), ...)` and pattern match inside each function clause. Today we are doing a mixture which is a bit unclear.

File: lib/phoenix/digester.ex [hunk 2]
```diff
@@ -185,35 +185,43 @@ defmodule Phoenix.Digester do
...
+          open <> quote_symbol <> digested_url(url, file, manifest, true) <> quote_symbol <> close
         nil ->
-          open <> digested_url(url, file, manifest) <> close
+          open <> digested_url(url, file, manifest, true) <> close
       end
     end)
   end
 
-  defp digested_url("/" <> relative_path, _file, manifest) do
+  @javascript_sourceMap_regex ~r{(//#\s*sourceMappingURL=)(\S+)}
```
Comment (josevalim) [inline lib/phoenix/digester.ex]: `@javascript_source_map_regex`?

File: lib/phoenix/digester.ex [hunk 3]
```diff
@@ -185,35 +185,43 @@ defmodule Phoenix.Digester do
...
+
+  defp digest_asset_references(:js, file, manifest) do
+    Regex.replace(@javascript_sourceMap_regex, file.content, fn _, sourceMapText, url ->
```
Comment (josevalim) [inline lib/phoenix/digester.ex]: The var should be `source_map_text`.

**Comment by josevalim** [comment]
@Szetty In my opinion it is also ok to digest the contents of the `.map` files. Can you think of any reason not to? Shouldn't we digest so we are able to point back to the proper JS file?

**Comment by Szetty** [comment]
thx for commenting @josevalim, there is absolutely no reason not to digest the contents of the .map files, just that I didn't know if we should open a new issue for this, to be able to track separate tasks, or I can implement it in this pull request, because they have some connection.

Talking about this from CONTRIBUTING.md : "Good pull requests - patches, improvements, new features - are a fantastic help. They should remain focused in scope and avoid containing unrelated commits."

**Comment by josevalim** [comment]
Yes, let's do two PRs. That would be better.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**